server.js

```diff
@@ -1,29 +1,50 @@
 import express from "express";
 import cors from "cors";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+import netflixData from "./data/netflix-titles.json";
+
 const port = process.env.PORT || 8080;
 const app = express();
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
+
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send("Netflix & chill!");
+});
+
+app.get("/stream", (req, res) => {
+  res.status(200).json(netflixData)
 });
 
+app.get("/stream/year/:year", (req, res) => {
+  const { year } = req.params;
+  const streamByYear = netflixData.filter((stream => stream.release_year === +year))
+  streamByYear
+    ? res.status(200).json({ data: streamByYear })
+    : res.status(400).json({ error: "Not found" })
+})
+
+app.get("/stream/title/:title", (req, res) => {
+  const { title } = req.params;
+  const streamByTitle = netflixData.find(stream => stream.title.toLowerCase() === title.toLowerCase())
+  streamByTitle
+    ? res.status(200).json({ data: streamByTitle })
+    : res.status(400).json({ error: "Not found" })
+})
+
+app.get("/stream/type/:type", (req, res) => {
+  const { type } = req.params;
+  const streamByTitle = netflixData.filter(stream => stream.type.toLowerCase() === type.toLowerCase())
+  streamByTitle
+    ? res.status(200).json({ data: streamByTitle })
+    : res.status(400).json({ error: "Not found" })
```

Comment:
I couldn´t make the error-handling work on the endpoints where you used filter() method. Tried changing a bit in your code and it worked if I changed the condition (streamByTitle on row 42 in this case) to "streamByTitle.length === 0" and the switch the places of the respons. Tried on the other filtered aswell and it worked. Not sure why, but it might have to do with that the filter method returns an array so it is the length that needs to be compared. Try it out if you like!